### PR TITLE
Remove Utils.FileExists(), Utils.DirectoryExists() and Utils.ItemExists() methods

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

Continue #6929

Cleanup code to replace Utils.FileExists(), Utils.DirectoryExists() and Utils.ItemExists() methods with .Net Core methods.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -2874,41 +2886,20 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
...
-                    errorRecord.ErrorDetails = errorDetails;
-
-                    WriteError(errorRecord);
-                    return;
+                    string error = StringUtil.Format(FileSystemProviderStrings.CannotRemoveItem, directory.FullName);
+                    Exception exception = new IOException(error, exc);
+                    WriteError(new ErrorRecord(exception, "DeleteJunctionFailed", ErrorCategory.WriteError, directory));
                 }
+
+                return;
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: `return` was within the catch block.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: @adityapatwardhan Could you please clarify why we should move the return in catch block to continue recursion removal?
Previous code was trying the recursion removal if only we can not remove reparse point _silently_ without any error and exception - is it real scenario?

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
Rebased to get latest fixes.

**Comment by iSazonov** [comment]
I update some commits:
- Fix typo `if ((int)item.Attributes != 1)`  ->  `if ((int)item.Attributes != -1)`
- Replace wrong pattern `isDirectory = attrs.HasFlag(FileAttributes.Directory);` with `isDirectory = exists && attrs.HasFlag(FileAttributes.Directory);`
- Fix CodeFactor issues.

Seems I fixed all Appveyor CI issues. I can not build Unix version locally so I haven't a progress with Travis CI issues.

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 2]
```diff
@@ -109,20 +109,36 @@ private static string NormalizePath(string path)
...
         private static FileSystemInfo GetFileSystemInfo(string path, ref bool isContainer)
         {
-            isContainer = false;
+            // We use 'FileInfo.Attributes' (not 'FileInfo.Exist')
+            // because we want to get exceptions
+            // like UnauthorizedAccessException or IOException.
+            FileSystemInfo fsinfo = new FileInfo(path);
+            var attr = fsinfo.Attributes;
+            var exists = (int)attr != -1;
+            isContainer = exists && attr.HasFlag(FileAttributes.Directory);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: I see code like this for checking if a path exists and whether it's directory at 8 places. I think keeping the redundant code in `Utils.ItemExists` is better. Thoughts?

**Comment by iSazonov** [comment]
Continue in #7129

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs (hunk 2 above)
Comment (daxian-dbw) [reply src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Thanks @iSazonov

**Pull request closed without merging**